deps/zlib.mk

```diff
@@ -0,0 +1,3 @@
+ZLIB_BB_URL_BASE := https://github.com/JuliaBinaryWrappers/Zlib_jll.jl/releases/download/Zlib-v$(ZLIB_VER)+$(ZLIB_BB_REL)
+ZLIB_BB_NAME := Zlib.v$(ZLIB_VER)
+$(eval $(call bb-install,zlib,ZLIB,false))
```

Comment:
I guess it's unclear to me—are we supposed to support USE_BINARYBUILDER=0 still?

Reply:
If you're compiling from source, I think it's reasonable to expect that the user will have `libz` installed.  I'm not sure what to do about `7z` though.

Reply:
To be a little more clear on the dilemma here; building `7z` is easy on all platforms except windows, where we are already forcing users to download binaries.

I think the right middle ground here is to allow the user to compile with `USE_BINARYBUILDER=0`, but note in `README.md` that Julia needs `7z` to use `Pkg`, and if you turn off BinaryBuilder, you're going to need to provide a `7z` on the host machine.   Does that sound reasonable?

Reply:
@StefanKarpinski pinging you since I think you'll have an opinion here.

Reply:
As discussed on the Pkg call, yes that sounds reasonable.